pkg/controller/elasticsearch/driver/driver.go

```diff
@@ -251,6 +252,11 @@ func (d *defaultDriver) Reconcile(ctx context.Context) *reconciler.Results {
 		return results.WithError(err)
 	}
 
+	// requeue if associations are defined but not yet configured
```

Comment:
for the sake of explicitness:

`//..., otherwise we may be in a situation where we deploy Elasticsearch Pods once, then change their spec a few seconds later once the association is configured`